packages/syft/src/syft/core/node/new/lib_service_registry.py

```diff
@@ -0,0 +1,345 @@
+        self.absolute_path = absolute_path
+
+        self.children: Dict[str, CMPBase] = dict()
+        if isinstance(children, list):
+            self.children = {f"{c.path}": c for c in children}
+        elif isinstance(children, dict):
+            self.children = children
+
+        for c in self.children.values():
+            if c.absolute_path is None:
+                c.absolute_path = f"{path}.{c.path}"
+
+        if text_signature is not None:
+            self.signature: Signature = _signature_fromstr(
+                inspect.Signature, obj, text_signature, True
+            )
+        else:
+            self.signature = None
+
+        self.is_built = False
+
+    def set_signature(self):
+        pass
+
+    def build(self):
+        if self.obj is None:
+            self.obj = import_from_path(self.absolute_path)
+
+        if self.signature is None:
+            self.set_signature()
+
+        child_paths = set([p for p in self.children.keys()])
+
+        for attr_name in getattr(self.obj, "__dict__", dict()).keys():
+            if attr_name not in LIB_IGNORE_ATTRIBUTES:
+                if attr_name in child_paths:
+                    child = self.children[attr_name]
+                else:
+                    try:
+                        attr = getattr(self.obj, attr_name)
+                    except Exception:
+                        continue
+                    child = self.init_child(
+                        self.obj,
+                        f"{self.path}.{attr_name}",
+                        attr,
+                        f"{self.absolute_path}.{attr_name}",
+                    )
+                if child is not None:
+                    child.build()
+                    self.children[attr_name] = child
+
+    def __getattr__(self, __name: str) -> Any:
+        if __name in self.children:
+            return self.children[__name]
+        else:
+            raise ValueError(f"property {__name} not defined")
+
+    def init_child(self, parent_obj, child_path: str, child_obj, absolute_path):
+        """Get the child of parent as a CMPBase object
+
+        Args:
+            parent_obj (_type_): parent object
+            child_path (_type_): _description_
+            child_obj (_type_): _description_
+
+        Returns:
+            _type_: _description_
+        """
+        parent_is_parent_module = CMPBase.parent_is_parent_module(parent_obj, child_obj)
+        if CMPBase.isfunction(child_obj) and parent_is_parent_module:
+            return CMPFunction(
+                child_path,
+                permissions=self.permissions,
+                obj=child_obj,
+                absolute_path=absolute_path,
+            )
+        elif inspect.ismodule(child_obj) and CMPBase.is_submodule(
+            parent_obj, child_obj
+        ):
+            ## TODO, we could register modules and functions in 2 ways:
+            # A) as numpy.float32 (what we are doing now)
```

Comment:
This was an issue in previous versions of syft and what we did was cache the paths during construction so that x.y and x.m.y resolve to the same thing.